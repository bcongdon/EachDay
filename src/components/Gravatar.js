import React, { Component } from 'react'
import { Image } from 'semantic-ui-react'
import { PropTypes } from 'prop-types'
import md5 from 'js-md5'

class Gravatar extends Component {
  getUrl () {
    const hash = md5(this.props.email)
    return `https://www.gravatar.com/avatar/${hash}?s=${this.props.size}&r=pg`
  }

  render () {
    return (
      <Image avatar={this.props.avatar} src={this.getUrl()} />
    )
  }
}

Gravatar.propTypes = {
  email: PropTypes.string.isRequired,
  size: PropTypes.number,
  avatar: PropTypes.bool
}

Gravatar.defaultProps = {
  size: 80,
  avatar: true
}

export default Gravatar
