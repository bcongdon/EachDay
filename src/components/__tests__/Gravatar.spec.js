import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Gravatar from '../Gravatar'

it('renders correctly', () => {
  const gravatar = shallow(
    <Gravatar email='foo@bar.com' />
  )
  expect(toJson(gravatar)).toMatchSnapshot()
})

it('renders correctly with custom size', () => {
  const gravatar = shallow(
    <Gravatar email='foo@bar.com' size={720} />
  )
  expect(toJson(gravatar)).toMatchSnapshot()
})
