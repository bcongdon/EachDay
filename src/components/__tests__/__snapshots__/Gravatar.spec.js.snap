// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  as="img"
  avatar={true}
  src="https://www.gravatar.com/avatar/f3ada405ce890b6f8204094deb12d8a8?s=80&r=pg"
  ui={true}
/>
`;

exports[`renders correctly with custom size 1`] = `
<Image
  as="img"
  avatar={true}
  src="https://www.gravatar.com/avatar/f3ada405ce890b6f8204094deb12d8a8?s=720&r=pg"
  ui={true}
/>
`;
